epilogue-processor/src/main/java/edu/wpi/first/epilogue/processor/LoggerGenerator.java

```diff
@@ -509,28 +515,30 @@ private List<String> generateDependencyLogging(Element element, TypeElement claz
     }
 
     if (dependencies.isEmpty()) {
-      return dependencyInvocations;
+      return null;
     }
 
-    // For each dependency, find the corresponding element and generate logging
+    // Collect valid dependency names
+    List<String> validDependencyNames = new ArrayList<>();
     for (DependsOn dependency : dependencies) {
       String dependencyName = dependency.value();
       Element dependencyElement = findDependencyElement(clazz, dependencyName);
 
       if (dependencyElement != null && isLoggable(dependencyElement)) {
-        // Generate logging call for the dependency
-        var handler = m_handlers.stream().filter(h -> h.isLoggable(dependencyElement)).findFirst();
-
-        if (handler.isPresent()) {
-          var logInvocation = handler.get().logInvocation(dependencyElement, clazz);
-          if (logInvocation != null) {
-            dependencyInvocations.add(logInvocation);
-          }
-        }
+        validDependencyNames.add(dependencyName);
       }
     }
 
-    return dependencyInvocations;
+    if (validDependencyNames.isEmpty()) {
+      return null;
+    }
+
+    // Create LogMetadata construction code
+    String dependencyList = validDependencyNames.stream()
+        .map(name -> "\"" + name + "\"")
+        .collect(java.util.stream.Collectors.joining(", "));
+    
+    return "new edu.wpi.first.epilogue.logging.LogMetadata(java.util.List.of(" + dependencyList + "))";
```

Comment:
Let's actually store metadata as fields in the generated loggers, eg:

```java
class ExampleLogger extends ClassSpecificLogger<Example> {
  private final LogMetadata fooMetadata = new LogMetadata(List.of("x", "y", "z"));

  @Override
  public void update(EpilogueBackend backend, Example object) {
    if (Epilogue.shouldLog(Logged.Importance.DEBUG)) {
      backend.log("x", object.x);
      backend.log("y", object.y);
      backend.log("z", object.z);
      backend.log("foo", object.foo, fooMetadata);
    }
  }
}
```

This would require two generation steps: one to _create_ the fields, and another to reference them in the logging calls.